storagenode/gracefulexit/worker.go

```diff
@@ -123,6 +125,41 @@ func (worker *Worker) Run(ctx context.Context, done func()) (err error) {
 				continue
 			}
 
+			if !bytes.Equal(originalHash.Hash, pieceHash.Hash) {
+				worker.log.Error("piece hash from new storagenode does not match", zap.Stringer("storagenode ID", addrLimit.Limit.StorageNodeId), zap.Stringer("satellite ID", worker.satelliteID), zap.Stringer("piece ID", pieceID))
```

Comment:
Log these errors a level higher rather than in these location, it adds a lot of additional code. Similarly, the mismatch must be detected and a "failed" message should be sent to the satellite.

Reply:
Could you elaborate on what you mean? Should I move the logging to a helper function so there aren't so many `Stringer`s in teh body here or something?

Reply:
The issue is that this single function does so many things it's really difficult to follow what is going on. It handles handling of the result, verification, scheduling communication etc.